pinot-common/src/main/java/org/apache/pinot/common/exception/QueryException.java

```diff
@@ -147,4 +148,24 @@ public static ProcessingException getException(ProcessingException processingExc
     copiedProcessingException.setMessage(errorType + ":\n" + errorMessage);
     return copiedProcessingException;
   }
+
+  /**
+   * Determines if a query-exception-error-code represents an error on the client side.
+   * @param errorCode  the error code from processing the query
+   * @return whether the code indicates client error or not
+   */
+  public static boolean isClientError(int errorCode) {
+    switch (errorCode) {
+      case QueryException.ACCESS_DENIED_ERROR_CODE:
+      case QueryException.JSON_COMPILATION_ERROR_CODE:
+      case QueryException.JSON_PARSING_ERROR_CODE:
+      case QueryException.QUERY_VALIDATION_ERROR_CODE:
+      case QueryException.PQL_PARSING_ERROR_CODE:
+      case QueryException.TOO_MANY_REQUESTS_ERROR_CODE: {
```

Comment:
remove {

Reply:
Please add a comment saying why we dont include BROKER_RESOURCE_MISSING_ERROR here as a client error.